src/bussinessLogic/ManageMap.java

```diff
@@ -56,38 +43,40 @@ public void setNewCharMap() {
 
         for (int i = 0; i < Constants.NUMBER_OF_LETTERS; i++) {
             charString = Character.toString(currentChar);
-            // charMap.put(charString, 0);
             map.put(charString, new LetterAttributes(0, 0));
-            dividerMap.put(charString, 0);
             currentChar++;
         }
     }
 
     /**
-     * sort the charMap in ascending order
+     * transfers the chars from map to rankingArray and
+     * arranges each char relative to its correlated reaction time
      */
-    public void sortCharMap() {
-        // generate a new, empty sortedMap
-        sortedMap = new LinkedHashMap<>();
-        // arrayList for storing and sorting the chars
+    public void rankChars() {
+
+        rankingArray = new char[map.size()];
+        int index = 0;
+        // arrayList for storing and sorting the chars relative to the typing speed
         ArrayList<Integer> charList = new ArrayList<>();
```

Comment:
Try sorting the characters without another `arraylist` like you did in this case.

Reply:
You're saying to sort the map directly?

Reply:
Exactly.